yb-voyager/src/tgtdb/yugabytedb.go

```diff
@@ -442,31 +479,34 @@ func (yb *TargetYugabyteDB) TruncateTables(tables []sqlname.NameTuple) error {
 	return nil
 }
 
-func (yb *TargetYugabyteDB) ImportBatch(batch Batch, args *ImportBatchArgs, exportDir string, tableSchema map[string]map[string]string) (int64, error) {
+/*
+ImportBatch function handles variety of cases for importing a batch into YugabyteDB
+1. Normal Mode - Importing a batch using COPY command with transaction.
+2. Fast Path Mode - Importing a batch using COPY command without transaction.
+3. Fast Path Recovery Mode - Importing a batch using COPY command without transaction but conflict handling.
+*/
+func (yb *TargetYugabyteDB) ImportBatch(batch Batch, args *ImportBatchArgs,
+	exportDir string, tableSchema map[string]map[string]string, isRecoveryCandidate bool) (int64, error) {
+
 	var rowsAffected int64
 	var err error
 	copyFn := func(conn *pgx.Conn) (bool, error) {
-		rowsAffected, err = yb.importBatch(conn, batch, args)
+		if args.IsFastPath() {
+			if !isRecoveryCandidate {
+				rowsAffected, err = yb.importBatchFast(conn, batch, args)
+			} else {
+				rowsAffected, err = yb.importBatchFastRecover(conn, batch, args)
+			}
+		} else {
+			rowsAffected, err = yb.importBatch(conn, batch, args)
```

Comment:
let's add a comment here explaining that in case of normal mode, we don't really have to care about recovery mode or not; since we run copy in a transaction.